### PR TITLE
Added option for no_export

### PR DESCRIPTION
**Pull request by Michanne**
Specify library name and version:  **assimp/5.0.x**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
This option was removed in the first PR adding assimp in CCI: https://github.com/conan-io/conan-center-index/pull/386/commits/92bb6999facac797b48efd72302120f36a445a0c

It doesn't add or remove any Assimp feature that you can't have with current options in the recipe. If you don't to disable all exports, just disable all individuals exports options.

If this option is really needed, there are some work to do in `package_id`, because `no_export=True` forces all exporters to False.

File: recipes/assimp/5.0.x/conanfile.py [hunk 1]
```diff
@@ -21,11 +21,13 @@ class Assimp(ConanFile):
         "shared": [True, False],
         "double_precision": [True, False],
         "fPIC": [True, False],
+        "no_export": [True, False],
     }
     default_options = {
         "shared": False,
         "double_precision": False,
         "fPIC": True,
+        "no_export": False,
```
Comment (SpaceIm) [inline recipes/assimp/5.0.x/conanfile.py:30]: all "export" like options (except no_export) must be deleted in `configure` method if `no_export=True`

**Comment by jgsogo** [comment]
Taking into account the review by @SpaceIm, PR https://github.com/assimp/assimp/pull/2759 and the future versions of this recipe:

 1. We can apply that patch to current `5.0.x` versions and keep options like they are right now
 2. We can merge this PR adding this new option (and removing the meaningless `~export_xxx` ones)
 3. We can introduce this new `no_export` option but, instead of removing the `~export` ones, we can force all of them to `False` if `no_export=True` and remove the `no_export` function itself from the packageID.

I prefer not to modify sources when possible, it makes releases in Conan different from the official ones. Thinking about future releases where https://github.com/assimp/assimp/pull/2759 is merged, option (1) will compute one packageID for `no_export=True` and a different one if you set all `export_*=False`, option (2) will compute the same packageID. 

So I need to ask: is there anything else done by `ASSIMP_BUILD_NO_EXPORT` other than disabling all the exports? is it creating a different ABI?

**Comment by SpaceIm** [comment]
> So I need to ask: is there anything else done by ASSIMP_BUILD_NO_EXPORT other than disabling all the exports? is it creating a different ABI?

No, `no_export=True` creates the same binary than disabling all individuals exports like options.

**Comment by danimtb** [comment]
I am closing this regarding the latest comments and the inactivity during the latest weeks. Thank you!

**Pull request closed without merging**